Cargo.toml

```
@@ -1,6 +1,6 @@
 [package]
 name = "fsrs"
-version = "0.3.0"
+version = "0.4.0"
```

```suggestion
version = "1.0.0"
```

This refactor introduces a lot of breaking changes. So we should upgrade the major version.